pkg/servicemonitor/servicemonitor.go

```diff
@@ -53,6 +54,26 @@ func (u *ServiceMonitor) UpdateServiceMonitorDeployment(template monitoringv1.Se
 	return nil
 }
 
+// Creates or Updates Service Monitor Deployment according to the template if enable of the hypershift
+func (u *ServiceMonitor) Hypershift_UpdateServiceMonitorDeployment(template rhobsv1.ServiceMonitor) error {
```

Comment:
I don't like that we need to duplicate the whole function just to use a different API group. Is there a smarter way to achieve this?

Reply:
Agreed, but any ideas? I'm not aware of a better way to do this at least, especially if these two API groups start diverging in the future.

Reply:
@anispate Do you think there is any better way here instead of a whole function to duplicate the api group ? Should we consider maybe outputting a structure that can hold several API groups as supported and have one function that knows how to create this ?

Reply:
API groups are types though, what we would like is:

"If HyperShift, then call `func UpdateServiceMonitorDeployment(template *rhobsv1)`, else `func UpdateServiceMonitorDeployment(template *monitoringv1)`, but I don't think this is possible.